DPPP/BDAIntervalBuffer.cc

```diff
@@ -93,11 +90,7 @@ namespace DP3 {
       }
 
       // Create the result buffer and fill it.
-      auto result = boost::make_unique<BDABuffer>(pool_size,
-                                                  enable_data,
-                                                  enable_flags,
-                                                  enable_weights,
-                                                  enable_full_res_flags);
+      auto result = boost::make_unique<BDABuffer>(pool_size, fields);
```

Comment:
As per google style code, only use `auto` with reason (e.g. templated code), not to prevent typing

Reply:
From the Google C++ style guide: "For example, you can assume that the return type of make_unique<Foo>() is obvious, but the return type of MyWidgetFactory() probably isn't."
The C++ Core Guidelines rule C.150 also say that for make_unique, 'auto' is better.

Reply:
Ok, sounds like you're right!

Reply:
Oh by the way, we don't use (and shouldn't use) all of the C++ Core Guidelines rules. I thought you were quoting the Google style guide. C150 is about make_unique, not about auto. Herb Sutter promotes using auto everywhere, which is conflicting with the (more pragmatically) google style that we follow, as are many of the other C++ core guidelines. make_unique is C++14 btw, and should not be used.